eventlog/api_test.go

```diff
@@ -143,3 +143,44 @@ func checkProof(t *testing.T, omni *omniledger.Service, key []byte, scID skipcha
 
 	return v[0].([]byte)
 }
+
+func TestClient_Search(t *testing.T) {
+	s := newSer(t)
+	leader := s.services[0]
+	defer s.close()
+
+	owner := darc.NewSignerEd25519(nil, nil)
+	c := NewClient(s.roster)
+	err := c.Init(owner, testBlockInterval)
+	require.Nil(t, err)
+
+	req := &SearchRequest{ID: c.ID}
+	resp, err := c.Search(req)
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, 0, len(resp.Events))
+
+	// Put 10 events in with 10 different timestamps that we can search on.
+	logCount := 10
+	tm0 := time.Now().UnixNano()
+	tm := tm0
+	for ct := 0; ct < logCount; ct++ {
+		tm += 1
+		_, err := c.Log(Event{Topic: "topic", Content: fmt.Sprintf("test event at time %v", tm), When: tm})
+		require.Nil(t, err)
+	}
+	leader.waitForBlock(c.ID)
+	leader.waitForBlock(c.ID)
+
+	req = &SearchRequest{ID: c.ID}
+	resp, err = c.Search(req)
+	require.Nil(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, 10, len(resp.Events))
+
+	req = &SearchRequest{ID: c.ID, From: tm0 + 3, To: tm0 + 8}
```

Comment:
Is this `From <= timestamp < To`? That might be worth a documentation in the `Search` api...

Reply:
It is documented in the SearchRequest type.